packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -466,6 +462,7 @@ export const run = async (context: $TSContext, resourceDefinition: $TSObject, re
     rollbackLambdaLayers(layerResources);
     throw new AmplifyFault('DeploymentFault', {
```

Comment:
@lazpavel I didn't test this, but I think this catch block always throws a fault when it might be catching an AmplifyError. Looks like a possible reporting bug.

Reply:
see this comment as it applies here as well, https://github.com/aws-amplify/amplify-cli/pull/11527#discussion_r1048742386, we report/print the deepest amplify exception in the chain, which is passed to this Fault